src/instruments/src/SD/Pages/Hyd/Hyd.tsx

```diff
@@ -0,0 +1,788 @@
+
+    const [elecRightFormat, setElecRightFormat] = useState('hide');
+    const [elecTriangleFill, setElecTriangleFill] = useState(0);
+    const [elecTriangleColour, setElecTriangleColour] = useState('white');
+
+    function setPressures(clearState = false) {
+        if (clearState) {
+            setPressureChart({ high: '', low: '', highValue: -1, lowValue: -1, ptuScenario: 'normal' });
+        } else if (yellowPressure > greenPressure) {
+            setPressureChart({
+                high: 'YELLOW',
+                low: 'GREEN',
+                highValue: yellowPressure,
+                lowValue: greenPressure,
+                ptuScenario: 'right-to-left',
+            });
+        } else {
+            setPressureChart({
+                high: 'GREEN',
+                low: 'YELLOW',
+                highValue: greenPressure,
+                lowValue: yellowPressure,
+                ptuScenario: 'left-to-right',
+            });
+        }
+    }
+
+    useEffect(() => {
+        setPtuScenario(pressureChart.ptuScenario);
+    }, [pressureChart]);
+
+    // PTU logic
+    useEffect(() => {
+        if (yellowElectricPumpStatus) {
+            setElecTriangleFill(1);
+            setElecTriangleColour(yellowPressure <= 1450 ? 'amber' : 'green');
+            setElecRightFormat(yellowPressure <= 1450 ? 'amber-line' : 'green-line');
+        } else {
+            setElecTriangleFill(0);
+            setElecTriangleColour('white');
+            setElecRightFormat('hide');
+        }
+
+        if (ptuAvailable && !yellowElectricPumpStatus) {
+            // The PTU valve has to be open and the yellow electric pump should not be on
+            const pressureDifferential = Math.abs(greenPressure - yellowPressure);
+            const maxPressure = Math.max(yellowPressure, greenPressure);
+            // const minPressure = Math.min(yellowPressure, greenPressure);
+            const negativePressureDifferential = pressureChart.low === 'GREEN' ? pressureChart.lowValue - yellowPressure : pressureChart.lowValue - greenPressure;
+            if (maxPressure < 1450 || (greenPressure > 2990 && yellowPressure > 2990)) {
+                setPressures(true);
+                setPtuActive(0);
+            } else if (pressureDifferential > 200 && maxPressure > 1450 && !ptuActive) {
+                console.log('Pressure differ > 200');
+                setPtuActive(1);
+                setPressures();
+            } else if (negativePressureDifferential <= -500 && ptuActive) {
+                console.log('Pressure differential is < -500');
+                setPressures(true);
+                setPtuActive(0);
+            }
+        } else if (ptuAvailable && yellowElectricPumpStatus && greenPressure <= 2990) {
+            setPtuScenario('right-to-left');
+            setPtuActive(1);
+        } else {
+            setPtuScenario(ptuAvailable ? 'normal' : 'PTU-off');
+        }
+    }, [greenPressure, yellowPressure, yellowElectricPumpStatus]);
```

Comment:
**thought**: I hope @crocket63 will provide enough data to not require these calculation in the presentation layer. We should strive to keep the React side as lean as possible.

Reply:
There's actually specific ecam logic that is not aware of the PTU state. If we go full simulation that logic might be ok to be here.

Reply:
If there's a true performance issue, it wouldn't be too difficult to process the exact same logic runing here on rust side just outputting the required ecam boolean A32NX_HYD_ECAM_GREEN_AVAILABLE  A32NX_HYD_ECAM_PTU_G_2_Y_ACTIVE etc....